lib/engine/game/g_18_india/game.rb

```diff
@@ -1104,27 +1110,27 @@ def visit_jewelery(route)
           route.all_hexes.include?(@jewlery_hex) ? ['JEWELRY'] : []
         end
 
-        def commodity_bonus(route, _stops = nil)
+        def commodity_bonus(route, _stops = nil, report_commodities: false)
           visited_names = (route.all_hexes.map(&:location_name) + visit_jewelery(route)).compact
           corporation = route.train.owner
           commodity_sources = visited_names & available_commodities(corporation)
-          return 0 unless commodity_sources.count.positive?
+          return (report_commodities ? [] : 0) unless commodity_sources.count.positive?
 
           revenue = 0
-          @round.commodities_used = []
+          commodities_used = []
           commodity_sources.each do |source|
             bonus = COMMODITY_BONUSES[source]
             LOGGER.debug { "GAME.commodity_bonus >> bonus: #{bonus}" }
-            if visited_names.intersect?(bonus['locations'])
+            if visited_names.intersect?(bonus[:locations])
               revenue += bonus['value'] || visited_names.map { |loc| SPICE_BONUSES[loc] || 0 }.max
```

Comment:
The `bonus` hash key needs to be changed to a label here as well.
```suggestion
              revenue += bonus[:value] || visited_names.map { |loc| SPICE_BONUSES[loc] || 0 }.max
```